### ISSUE TITLE
rtl8822be device not ready

### ISSUE THREAD

**Issue by MikTur** [lwfinger/rtlwifi_new#415]
Hi all,
Recently have changed the motherboard for my pc and found that if has the rtl8822be wifi card.
Have followed the instructions in order to build the driver and install it.
My host machine is equipped with debian linux using kernel version 4.16
At every boot however the Network Manager is not able to get the card to be used as it seems to be not ready.
Floating on the web I have noticed that few people have resolved a similar issue by creating a rtl8822be.conf file in the modprobe folder with the aspm=0 option written in it...but such a move did not help.
When i run dmesg I get this wired stuff for my rtl8822be module:

[    4.568903] rtl8822be: Using firmware rtlwifi/rtl8822befw.bin
[    4.571265] rtl8822be 0000:04:00.0: firmware: direct-loading firmware rtlwifi/rtl8822befw.bin
[    4.574172] rtl8822be 0000:04:00.0 wlp4s0: renamed from wlan0
[    6.984776] rtl8822be: rtl8822b_halmac_cb_write_data_rsvd_page polling beacon fail
[   19.296901] rtl8822be: halmac_init_hal failed
[   19.725527] rtl8822be: halmac_init_hal failed
[   19.725928] rtl8822be: halmac_init_hal failed
[   30.003336] rtl8822be: halmac_init_hal failed
[   30.003486] rtl8822be: halmac_init_hal failed
[   40.005762] rtl8822be: halmac_init_hal failed
[   40.005901] rtl8822be: halmac_init_hal failed
[   50.207652] rtl8822be: halmac_init_hal failed
[   50.207808] rtl8822be: halmac_init_hal failed

Any help will be so much appreciated.
Thanks.

**Comment by lwfinger** [comment]
The "aspm=0" option has fixed every case like this so far. Are you sure you created the options file OK? The line should say "options rtl8822be aspm=0".

**Comment by MikTur** [comment]
Hi yep,
the conf filename is /etc/modprobe.d/rtl8822be.conf and its content is just the one you have wrote:
options rtl8822be aspm=0

However I got blacklisted in another conf file the rtl8xxxu driver maybe is that issue?
blacklist rtl8xxxu
options rtl8xxxu modeset=0

**Comment by lwfinger** [comment]
No.The "u" at the end of rtl8xxxu means that it handles USB devices. Your RTL8822BE is a PCIe device.

**Comment by MikTur** [comment]
Maybe it is something related to the mb bios...I got an Asus ROG Strix x470-I which to be honest with the previous 0631 bios version I got the wifi adapter recognized only once, at the first start. - this was observable in Windows as well.
Now with the 0931 bios version I can see the wifi adapter being settled in place at every sustem launch but can not make it work :/
Is there any other dump that I can attach to help us investigate this issue?
Thanks

**Comment by MikTur** [comment]
Iwfinger, guys, even if I appreciate your efforts to make the new realtek wifi adapters work I am probably going to open the "wifi go" enclosure on my mb and replace that crap with an old stable INTEL AC 7260 card...that crap does seem to work on Windows either.

**Comment by MikTur** [comment]
Have upgraded the kernel to version 4.18, recompiled the rtlwifi driver, installed and configured it and nothing, the issue still exists.
Thank you Iwfinger but at this point I am going to try with another wifi card.

**Comment by MikTur** [comment]
I did my last try this morning...I cloned the old repo: the https://github.com/synthtc/rtlwifi-next one.
I modified the timer init calls according to the new API provided since kernel 4.17 version --> timer_setup()
I also have tweaked the ASPM init handler for the 8822be module ( the init handler signature is: void rtl8822be_init_aspm_vars(struct ieee80211_hw *hw) )
by letting also the pcie, pci bridge and the rest of the infrastructures to have the ASPM set to zero(0)
but nothing, even this driver version does not want to make this bloody adapter work.
ASPM or not the 8822be driver seems to do not work in every hw setup.
Leaving the battle :)
